### ISSUE TITLE
feat[dev]: ✨ implement A11y improvements to sd-switch

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#1488]
## Description
This task describes the Storybook implementation for the sd-switch component as part of the A11y improvements outlined in Epic https://github.com/solid-design-system/solid/issues/1465.

A11y improvements for sd-switch can be found here: [sd-switch Notes](https://www.notion.so/matuzo/sd-switch-10ef3f6e1bc0801bbb78da4faa0ab4c9).

---------------------------------

## Comment
**Problem:** Invalid checkboxes don't indicate that they're invalid.
**Solution:** Add and toggle aria-invalid="true" on the input element.

**Problem:** Error messages are semantically not connected to the respective input element and are only announced when the user unticks a required checkbox.
**Solution:** Instead of using a live region for the error message, connect it with the affected input field. Screen readers then announce the message automatically when the user focuses the checkbox. ( -- take a look at #1619 for any overlap. Reach out to @Vahid1919 if you have any questions)

> This problem applies to most of the form elements and the solution should be addressed in the respective forms element tickets as well.

**Problem:** sd-switch uses a "checkbox" role and therewith offers 3 states. As we only have 2 states with the sd-switch, we need to correct this => see 2nd note below
**Solution:** Either set a "switch" role or refactor the component to use a button => needs to be evaluated => pls adopt estimation.

## Notes
- Disabled switches: See [@sd-button](https://www.notion.so/matuzo/sd-button-104f3f6e1bc0800b9f72d2a046316ec3?pvs=25). Everything I wrote there about disabling form elements applies here as well.
- Your switch looks like a switch, but semantically it's a checkbox. It has two states (on/off), although its role (checkbox) suggests that there are three states (checked/unchecked/indeterminate). If toggling the switch doesn't have an immediate effect, you can leave it as is. If toggling it executes JS immediately, consider using a button instead. For more details, read Adrian Roselli's three articles about that topic.

## Best Practices
- Avoid switches. Many users have hard time understanding them.

## Additional resources
- https://adrianroselli.com/2019/03/under-engineered-toggles.html
- https://adrianroselli.com/2021/10/switch-role-support.html
- https://adrianroselli.com/2019/08/under-engineered-toggles-too.html

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
